docling_ibm_models/tableformer/data_management/tf_predictor.py

```diff
@@ -563,56 +568,66 @@ def multi_table_predict(
+                indexing_end_cols = []  # Index of original end col IDs (not indexes)
+                indexing_start_rows = (
+                    []
+                )  # Index of original start row IDs (not indexes)
+                indexing_end_rows = []  # Index of original end row IDs (not indexes)
+
+                # First, collect all possible predicted IDs, to be used as indexes
+                # ID's returned by Tableformer are sequential, but might contain gaps
+                for tf_response_cell in tf_responses:
+                    start_col_offset_idx = tf_response_cell["start_col_offset_idx"]
+                    end_col_offset_idx = tf_response_cell["end_col_offset_idx"]
+                    start_row_offset_idx = tf_response_cell["start_row_offset_idx"]
+                    end_row_offset_idx = tf_response_cell["end_row_offset_idx"]
+
+                    # Collect all possible col/row IDs:
+                    if start_col_offset_idx not in indexing_start_cols:
+                        indexing_start_cols.append(start_col_offset_idx)
+                    if end_col_offset_idx not in indexing_end_cols:
+                        indexing_end_cols.append(end_col_offset_idx)
+                    if start_row_offset_idx not in indexing_start_rows:
+                        indexing_start_rows.append(start_row_offset_idx)
+                    if end_row_offset_idx not in indexing_end_rows:
+                        indexing_end_rows.append(end_row_offset_idx)
+
+                indexing_start_cols.sort()
+                indexing_end_cols.sort()
+                indexing_start_rows.sort()
+                indexing_end_rows.sort()
+
+                # After this - put actual indexes of IDs back into predicted structure...
+                for tf_response_cell in tf_responses:
+                    tf_response_cell["start_col_offset_idx"] = (
+                        indexing_start_cols.index(
+                            tf_response_cell["start_col_offset_idx"]
+                        )
+                    )
+                    tf_response_cell["end_col_offset_idx"] = (
+                        tf_response_cell["start_col_offset_idx"]
+                        + tf_response_cell["col_span"]
+                    )
+                    tf_response_cell["start_row_offset_idx"] = (
+                        indexing_start_rows.index(
+                            tf_response_cell["start_row_offset_idx"]
+                        )
+                    )
+                    tf_response_cell["end_row_offset_idx"] = (
+                        tf_response_cell["start_row_offset_idx"]
+                        + tf_response_cell["row_span"]
+                    )
+                # Counting matched cols/rows from actual indexes (and not ids)
+                predict_details["num_cols"] = len(indexing_end_cols)
+                predict_details["num_rows"] = len(indexing_end_rows)
```

Comment:
if the sorting is disabled, we still need to populate `predict_details["num_cols"]` and `predict_details["num_rows"]` through another logic.

Reply:
Good point, added second commit with those fields populated straight from prediction OTSL in case re-sorting of cols/rows were skipped.